File: clients/ruby/model/colored_vertex.rb
require_relative 'vec2_float'
require_relative 'color_float'
class ColoredVertex
    attr_accessor :position
    attr_accessor :color
    def initialize(position, color)
        @position = position
        @color = color
    end
    def self.read_from(stream)
        position = Vec2Float.read_from(stream)
        color = ColorFloat.read_from(stream)
        ColoredVertex.new(position, color)
    end
    def write_to(stream)
        @position.write_to(stream)
        @color.write_to(stream)
    end
end

File: clients/haskell/src/RAIC/Model/ColoredVertex.hs
module RAIC.Model.ColoredVertex where

import           GHC.Generics          (Generic)
import           RAIC.Model.ColorFloat (ColorFloat)
import           RAIC.Model.Vec2Float  (Vec2Float)
import           RAIC.Utils.Trans      (Trans)

data ColoredVertex = ColoredVertex {
  position :: Vec2Float,
  color    :: ColorFloat
} deriving (Generic, Show)

instance Trans ColoredVertex
